### PR TITLE
[AIRFLOW-2102] Add custom_args to Sendgrid personalizations

### PR DESCRIPTION
**Pull request by ms32035**
Make sure you have checked _all_ steps below.

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
    - https://issues.apache.org/jira/browse/AIRFLOW-2102

### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:
   See linked Jira issue

### Tests
- [X] My PR adds the following unit tests:
`test_send_email_sendgrid_correct_email_custom_args`

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [X] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/utils/sendgrid.py [hunk 1]
```diff
@@ -63,6 +64,12 @@ def send_email(to, subject, html_content, files=None,
     mail.add_personalization(personalization)
     mail.add_content(Content('text/html', html_content))
 
+    # Add custom_args to personalization if present
+    pers_custom_args = kwargs.get('personalization_custom_args', None)
+    if isinstance(pers_custom_args, dict):
```
Comment (fenglu-db) [inline airflow/contrib/utils/sendgrid.py:69]: nit: should this be if pers_custom_args?

**Review by fenglu-db — COMMENTED** [review]
LGTM, thanks for the change.

File: airflow/contrib/utils/sendgrid.py (hunk 1 above)
Comment (ms32035) [reply airflow/contrib/utils/sendgrid.py:69]: My arbitrary decision to ignore invalid parameters instead of letting error occur later on

**Comment by ms32035** [comment]
Thanks, should we target the merge for 1.10?

**Pull request closed without merging**